spec/BowlingGameSpec.js

```diff
@@ -0,0 +1,97 @@
+describe('bowling game logic', function(){
+  var game;
+
+  beforeEach(function(){
+    game = new BowlingGame();
+  });
+
+  it('starts with score = 0', function() {
+    expect(game.score).toEqual(0);
+  });
+
+  it('has 10 frames', function() {
+    expect(game.allFrames.length).toEqual(10);
+  });
+
+  it('can roll a ball', function() {
+    game.roll(2);
+    expect(game.score).toEqual(2);
+  });
+
+  it('can only take rolls that are 10 or less', function() {
+    expect(function() { game.roll(11) } ).toThrow(new Error ("Illegal throw"));
+    expect(game.score).toEqual(0);
+  })
+
+  it('a roll in input into the frame scores', function() {
+    game.roll(5);
+    game.roll(4);
+    expect(game.allRolls[0]).toEqual([5,4]);
+  });
+
+  it('can only roll twice before moving onto another frame', function() {
+    game.roll(2);
```

Comment:
It would be really nice if you had a helper method for doing a succession of rolls. So you could do:
```javascript
rolls([2, 2, 2, 2]);